assets/src/blocks/components/not-post-type-check/index.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * External Imports
+ */
+import { some, castArray } from 'lodash';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * A component with renders its own children oly if the current editor post type
+ * is not one of the given `excludedPostTypeSlugs` prop.
+ *
+ * @param {?Object} postType
+ * @param {WPElement} children
+ * @param {(string|string[])} excludedPostTypeSlugs
+ * @return {WPElement|null} Rendered element or null.
+ * @constructor
+ */
+export function NotPostTypeCheck( {
+	postType,
+	children,
+	excludedPostTypeSlugs,
+} ) {
+	let isExcluded = false;
+	if ( postType ) {
+		isExcluded = some(
+			castArray( excludedPostTypeSlugs ),
+			( type ) => postType.slug === type
+		);
+	}
+	if ( isExcluded ) {
+		return null;
+	}
+
+	return children;
+}
+
+export default withSelect( ( select ) => {
+	const { getEditedPostAttribute } = select( 'core/editor' );
+	const { getPostType } = select( 'core' );
+	return {
+		postType: getPostType( getEditedPostAttribute( 'type' ) ),
+	};
+} )( NotPostTypeCheck );
```

Comment:
I'm still not sure I'm happy with this though.  Here's why:

- Add-ons might still need to reference some of these editor specific components (and they currently aren't exposed for that).

There's some potential options with moving things around:

1. I _could_ move this into `src/components/editor` but then that means we potentially load these unusable components in non-editor contexts (but it does exposed them for add-ons)
2. I could put these into `src/editor/components` but then how do we do dependencies?  `eejs.editor` will probably use blocks, but then blocks will need these components!
3. Alternative to the previous is putting these into `src/editor/components` but then we expose that on a new script building them to `eejs.editorComponents`.  Then we can just make that a dependency for the blocks script?
4. I could break out `eventespresso-core-blocks` so it is built and exposed as an object on `eejs`.  That way we can expose these components on `eejs.blockComponents` (or something like that)?  Then existing dependencies can still exist the way they are now.

I think I like the idea of option number three.  Thoughts?

Reply:
After discussing in slack with @tn3rb, I'm going to move this new component into `src/editor/hocs` and expose on `eejs.editorHocs`.  Then in the future that leaves room for eventually having `src/editor/components` and exposing those on `eejs.editorComponents`.  The reason for the separate bundles is because it's unlikely editor hocs will change as much as editor components so this allows them to be browser cached differently.